### PR TITLE
fix: stickiness

### PR DESCRIPTION
**Pull request by andreas-unleash**
- Fixes project model/store
- Fixes UI sync issues when switching stickiness 

## About the changes

Closes #

### CODE REVIEW

**Comment by andreas-unleash** [comment]
null checks failing is expected

File: frontend/src/component/feature/FeatureView/FeatureVariants/FeatureEnvironmentVariants/EnvironmentVariantsModal/EnvironmentVariantsModal.tsx [hunk 1]
```diff
@@ -308,14 +314,13 @@ export const EnvironmentVariantsModal = ({
     }, [apiPayload.error]);
 
     const handleClose = () => {
-        updateStickiness(defaultStickiness).catch(console.warn);
+        updateStickiness(defaultStickiness).then();
```
Comment (kwasniew) [inline frontend/src/component/feature/FeatureView/FeatureVariants/FeatureEnvironmentVariants/EnvironmentVariantsModal/EnvironmentVariantsModal.tsx:317]: empty then left

File: frontend/src/component/feature/FeatureView/FeatureVariants/FeatureEnvironmentVariants/EnvironmentVariantsModal/EnvironmentVariantsModal.tsx [hunk 2]
```diff
@@ -296,7 +302,7 @@ export const EnvironmentVariantsModal = ({
     };
 
     const onStickinessChange = (value: string) => {
-        updateStickiness(value).catch(console.warn);
+        updateStickiness(value).then();
```
Comment (kwasniew) [inline frontend/src/component/feature/FeatureView/FeatureVariants/FeatureEnvironmentVariants/EnvironmentVariantsModal/EnvironmentVariantsModal.tsx:305]: empty then left

File: frontend/src/component/feature/FeatureView/FeatureVariants/FeatureEnvironmentVariants/EnvironmentVariantsModal/EnvironmentVariantsModal.tsx [hunk 3]
```diff
@@ -273,7 +273,13 @@ export const EnvironmentVariantsModal = ({
         isChangeRequestConfigured(environment?.name || '') &&
         uiConfig.flags.crOnVariants;
 
-    const stickiness = variants[0]?.stickiness || defaultStickiness;
+    const stickiness = useMemo(() => {
+        if (!loading) {
+            return variants[0]?.stickiness || defaultStickiness;
+        }
+        return '';
+    }, [loading, defaultStickiness]);
```
Comment (kwasniew) [inline frontend/src/component/feature/FeatureView/FeatureVariants/FeatureEnvironmentVariants/EnvironmentVariantsModal/EnvironmentVariantsModal.tsx:281]: shouldn't variants[0] be on the list of memo deps? We're breaking exhaustive deps check rule here.

**Review by kwasniew — APPROVED** [review]
Non blocking approve with a few comments left

File: frontend/src/component/feature/FeatureView/FeatureVariants/FeatureEnvironmentVariants/EnvironmentVariantsModal/EnvironmentVariantsModal.tsx (hunk 3 above)
Comment (andreas-unleash) [reply frontend/src/component/feature/FeatureView/FeatureVariants/FeatureEnvironmentVariants/EnvironmentVariantsModal/EnvironmentVariantsModal.tsx]: yes - added it in, thanks

**Pull request merged**